### PR TITLE
Adds tip info to confirmation dialog

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/brave/brave-browser/issues/3023

UI implementation: https://github.com/brave/brave-ui/pull/353

Non-Recurring confirmation:
screen shot 2019-01-23 at 11 36 24 pm

Recurring confirmation:
screen shot 2019-01-23 at 11 37 06 pm

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
1. Start Brave from a fresh profile, enable Rewards, accept a grant
2. Make a recurring tip to a site
3. Confirm that the confirmation overlay displays info per design
4. Make a non-recurring tip to a site
5. Confirm that the confirmation overlay displays info per design
6. Do this a few times staggering different domains and ensure info is accurate

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_rewards/resources/donate/reducers/donate_reducer.ts [hunk 1]
```diff
@@ -45,13 +48,17 @@ const publishersReducer: Reducer<RewardsDonate.State> = (state: RewardsDonate.St
     case types.ON_DONATE:
       {
         if (state.publisher && state.publisher.publisherKey && payload.amount > 0) {
+          let amount = parseInt(payload.amount, 10)
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/donate/reducers/donate_reducer.ts]: contribution amount could be 7.5 as well, so we can't do int here

Comment (NejcZdovc) [reply components/brave_rewards/resources/donate/reducers/donate_reducer.ts]: can you please adjust the code, previous code was wrong

File: components/brave_rewards/resources/donate/reducers/donate_reducer.ts [hunk 2]
```diff
@@ -62,7 +69,20 @@ const publishersReducer: Reducer<RewardsDonate.State> = (state: RewardsDonate.St
...
     case types.ON_RECURRING_DONATIONS:
       state = { ...state }
-      state.recurringList = action.payload.list
+      const recurringDonations = action.payload.recurringDonations
+      const publisherKey = state.publisher && state.publisher.publisherKey
+
+      if (publisherKey) {
+        let recurringDonation
+
+        recurringDonations.map((donation: RewardsDonate.RecurringDonation) => {
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/donate/reducers/donate_reducer.ts]: can we use find here instead of map?

Comment (ryanml) [reply components/brave_rewards/resources/donate/reducers/donate_reducer.ts]: Done

**Pull request merged**